rules/codelingo/effective-go/reuse-variable-name-type-switch/codelingo.yaml

```diff
@@ -10,11 +10,46 @@ tenets:
   - name: reuse-variable-name-type-switch
     actions:
       codelingo/docs:
-        title: Reuse the variable name in a type switch
+        title: Reuse Variable Name In A Type Switch
         body: |
-          If the switch declares a variable in the expression, the variable will have the corresponding type in each clause. It's also idiomatic to reuse the name in such cases, in effect declaring a new variable with the same name but a different type in each case.
+          If the switch declares a variable in the expression, 
+          the variable will have the corresponding type in each clause. 
+          It's also idiomatic to reuse the name in such cases, in effect 
+          declaring a new variable with the same name but a different 
+          type in each case.
+
+          ## Bad code
+          ```golang
+          // Print prints a different statement for int, string or default input types
+          func Print(i interface{}) {
+              switch v := i.(type) {
+              case int:
+                fmt.Printf("Twice %v is %v\n", v, v*2)
+              case string:
+                fmt.Printf("%q is %v bytes long\n", v, len(v))
+              default:
+                fmt.Printf("I don't know about type %T!\n", v)
+              }
+          }
+          ```
+
+          ## Good code
+          ```golang
+          // Print prints a different statement for int, string or default input types
+          func Print(i interface{}) {
+              switch i := i.(type) {
```

Comment:
```suggestion
              switch i := i.(type) { // <-- reuse of variable i here.
```

Reply:
I've added a comment here, just because this one is relatively hard to visually spot-the-difference. Loving the realistic code samples, though! 👍